### ISSUE TITLE
Palindrome instruction unclear

### ISSUE THREAD

**Issue by mayppong** [exercism/xelixir#68]
The palindrome instruction in the README seems to fall short compares to instructions from other exercises I've seen so far.

1. The description of palindrome product could use a few more examples. It should also clarify that there may be more than 1 solution to the product like with `9 => [[1, 9], [3, 3]]`, and that all single digits were considered palindrome products. Perhaps a link to the wiki page would be good?
2. The expected resulting format wasn't given. I had figured out after reading the test suite that it was doing `Dict.keys` on my results.
3. The test was not obvious either. The first test was looking at the last element in my result checking that I had return palindrome value for 9 but the need the format I should return so that somehow I ended up with key 9 being the last element wasn't obvious.

Just wanted to drop a note in case someone else agrees and it's worthwhile. I could probably do it sometimes in the future when I'm not too busy too.

Thanks!

**Comment by kytrinyx** [comment]
It's always worthwhile improving the exercises!

The README uses the same description across all the language tracks, which means that we're intentionally vague when it comes to actual data structures, since these vary quite broadly. https://github.com/exercism/x-common/blob/master/palindrome-products.md

However, having more examples and a link to the wiki page would be great.

I'll let the others comment on the suggestion for improving the test, since I don't write elixir.

**Comment by dalexj** [comment]
I just did this exercise and it took me awhile to figure this out as well. I came here to post an issue to discuss with anyone and this was already here. +1 to making this more clear somehow. I think the first test for this should just show the structure of the return data, for example:

```elixir
  test "structure from first 2 factors" do
    palindromes = Palindromes.generate(2) |> IO.inspect
    expected = %{
      1 => [[1, 1]],
      2 => [[1, 2]],
      4 => [[2, 2]]
    }
    assert palindromes == expected
  end
```

we might want something more but I agree with @mayppong on the return values of this exercise were very confusing.
